### DISCUSSION TITLE
V3 Rest API: summary vs detailed representation

### DISCUSSION THREAD

**Discussion by jrhone** [community/community#24899] [Apps, API and Webhooks]
<p>Hello! I’m using the v3 rest api.</p>
<p>My question’s about summary vs detailed representations of an object as documented on <a href="https://developer.github.com/v3/" rel="nofollow noopener">https://developer.github.com/v3/</a></p>
<p>Summary: When you fetch a list of resources, the response includes a subset of the attributes for that resource.</p>
<p>Detailed: When you fetch an individual resource, the response typically includes all attributes for that resource.</p>
<p>Do you have it documented which list apis return summary data and which return detailed data?</p>
<p>Get-repos returns a list of summary but it seems as if get-labels, since it has no computationally expensive fields, returns a list of detailed (unless I’m mistaken).</p>
<p>How can I know which list apis return a detailed representation?</p>
<p>Thanks</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>No, we don’t have a definitive list of which V3 APIs will return all possible attributes for an object and which won’t. You’ll probably have to do some experimentation to determine exactly which APIs have the information you need for a specific use.</p>
<p>On the other hand, this is, as I understand it, one of the major benefits of the V4 GraphQL-based API. When you query for objects, you can specify which attributes you need returned. You may want to do some experimentation there to see if the GraphQL API will work better for your needs.</p>
